Patch macro in logs that isn't handled by ./configure

diff --git a/libavcodec/cbs.c b/libavcodec/cbs.c
index b26e39eab4..c83a87e2cb 100644
--- a/libavcodec/cbs.c
+++ b/libavcodec/cbs.c
@@ -367,7 +367,7 @@ static int cbs_write_unit_data(CodedBitstreamContext *ctx,
         if (ret < 0) {
             av_log(ctx->log_ctx, AV_LOG_ERROR, "Unable to allocate a "
                    "sufficiently large write buffer (last attempt "
-                   "%"SIZE_SPECIFIER" bytes).\n", ctx->write_buffer_size);
+                   "%zu bytes).\n", ctx->write_buffer_size);
             return ret;
         }
     }
diff --git a/libavcodec/cbs_av1.c b/libavcodec/cbs_av1.c
index 98d745f494..6ccc3b624e 100644
--- a/libavcodec/cbs_av1.c
+++ b/libavcodec/cbs_av1.c
@@ -689,7 +689,7 @@ static int cbs_av1_split_fragment(CodedBitstreamContext *ctx,
 
     if (INT_MAX / 8 < size) {
         av_log(ctx->log_ctx, AV_LOG_ERROR, "Invalid fragment: "
-               "too large (%"SIZE_SPECIFIER" bytes).\n", size);
+               "too large (%zu bytes).\n", size);
         err = AVERROR_INVALIDDATA;
         goto fail;
     }
@@ -740,7 +740,7 @@ static int cbs_av1_split_fragment(CodedBitstreamContext *ctx,
         if (obu_header.obu_has_size_field) {
             if (get_bits_left(&gbc) < 8) {
                 av_log(ctx->log_ctx, AV_LOG_ERROR, "Invalid OBU: fragment "
-                       "too short (%"SIZE_SPECIFIER" bytes).\n", size);
+                       "too short (%zu bytes).\n", size);
                 err = AVERROR_INVALIDDATA;
                 goto fail;
             }
@@ -757,7 +757,7 @@ static int cbs_av1_split_fragment(CodedBitstreamContext *ctx,
 
         if (size < obu_length) {
             av_log(ctx->log_ctx, AV_LOG_ERROR, "Invalid OBU length: "
-                   "%"PRIu64", but only %"SIZE_SPECIFIER" bytes remaining in fragment.\n",
+                   "%"PRIu64", but only %zu bytes remaining in fragment.\n",
                    obu_length, size);
             err = AVERROR_INVALIDDATA;
             goto fail;
@@ -836,7 +836,7 @@ static int cbs_av1_read_unit(CodedBitstreamContext *ctx,
     } else {
         if (unit->data_size < 1 + obu->header.obu_extension_flag) {
             av_log(ctx->log_ctx, AV_LOG_ERROR, "Invalid OBU length: "
-                   "unit too short (%"SIZE_SPECIFIER").\n", unit->data_size);
+                   "unit too short (%zu).\n", unit->data_size);
             return AVERROR_INVALIDDATA;
         }
         obu->obu_size = unit->data_size - 1 - obu->header.obu_extension_flag;
diff --git a/libavcodec/cbs_vp9.c b/libavcodec/cbs_vp9.c
index ff99fe32fb..0e94c336a5 100644
--- a/libavcodec/cbs_vp9.c
+++ b/libavcodec/cbs_vp9.c
@@ -415,7 +415,7 @@ static int cbs_vp9_split_fragment(CodedBitstreamContext *ctx,
         }
         if (pos + index_size != frag->data_size) {
             av_log(ctx->log_ctx, AV_LOG_WARNING, "Extra padding at "
-                   "end of superframe: %"SIZE_SPECIFIER" bytes.\n",
+                   "end of superframe: %zu bytes.\n",
                    frag->data_size - (pos + index_size));
         }
 
